tests/test_gridsearch.py

```diff
@@ -130,9 +130,8 @@ def test_grid_search_no_score(self):
             grid_search.score(X, y), grid_search_no_score.score(X, y))
 
         # giving no scoring function raises an error
-        grid_search_no_score = TuneGridSearchCV(clf_no_score, {"C": Cs})
         with self.assertRaises(TypeError) as exc:
-            grid_search_no_score.fit([[1]])
+            grid_search_no_score = TuneGridSearchCV(clf_no_score, {"C": Cs})
```

Comment:
Why does the error raise earlier when instantiating `TuneGridSearchCV`?

Reply:
Because we have `_check_multimetric_scoring` in the constructor now instead of down in fit. I think this is necessary though; makes more sense because we need to know if it's multimetric ahead of time to ensure `refit` is set properly.